evm/trie/trie_test.go

```diff
@@ -0,0 +1,365 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package trie
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethdb/leveldb"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+var (
+	defaultTriesToStore = 3
+)
+
+func createNewTxTries(numHistoricalTries int) *TxTries {
+	return NewTxTries(numHistoricalTries)
+}
+
+func createTempDB() *leveldb.Database {
+
+	diskdb, err := leveldb.New("./temp-database", 256, 0, "")
+	if err != nil {
+		panic(fmt.Sprintf("unable to create testing database: %v", err))
+	}
+	return diskdb
+}
+
+func deleteTempDB() {
+	os.RemoveAll("./temp-database")
+}
+
+func addTrie(txTries *TxTries, root common.Hash, transactions []common.Hash, db *leveldb.Database) error {
+
+	if db == nil {
+		db = createTempDB()
+	}
+
+	if transactions == nil {
+		transactions = make([]common.Hash, 0)
+	}
+
+	_, err := txTries.AddTrie(root, db, transactions)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
```

Comment:
The implicit creation of the DB by passing nil seems convoluted. Would be easier to read the tests if you just create a DB and pass it in.